libmscore/cmd.cpp

```diff
@@ -2498,7 +2498,8 @@ void Score::cmdInsertClef(Clef* clef, ChordRest* cr)
 
 void Score::cmdAddGrace (NoteType graceType, int duration)
       {
-      for (Element* e : selection().elements()) {
+      QList<Element*> copyOfElements = selection().elements();
```

Comment:
It might be better to declare it as `const QList<Element*>` to prevent unnecessary actual copying of `QList` in the loop below if nothing changes, see [`qAsConst` documentation](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) for more details.